java-frontend/src/main/java/org/sonar/java/UCFGJavaVisitor.java

```diff
@@ -276,21 +276,26 @@ private void buildCall(Tree element, UCFGBuilder.BlockBuilder blockBuilder, Iden
       buildPlusAssignmentInvocation(blockBuilder, idGenerator, assignmentExpressionTree);
     } else if (element.is(ARRAY_ACCESS_EXPRESSION) && !element.parent().is(PLUS_ASSIGNMENT, ASSIGNMENT)) {
       // PLUS_ASSIGNMENT and ASSIGNMENT might imply an array set, otherwise an array access is always a get
-      Expression.Variable getValue = variableWithId(idGenerator.newId());
-      Expression array = idGenerator.lookupExpressionFor(((ArrayAccessExpressionTree)element).expression());
-      blockBuilder.assignTo(getValue, arrayGet(array), location(element));
-      idGenerator.varForExpression(element, getValue.id());
+      ArrayAccessExpressionTree arrayAccessExpressionTree = (ArrayAccessExpressionTree) element;
+      if (isObject(arrayAccessExpressionTree.symbolType())) {
+        Expression.Variable getValue = variableWithId(idGenerator.newId());
+        Expression array = idGenerator.lookupExpressionFor((arrayAccessExpressionTree).expression());
+        blockBuilder.assignTo(getValue, arrayGet(array), location(element));
+        idGenerator.varForExpression(element, getValue.id());
+      }
     }
   }
 
   private void buildNewArrayInvocation(BlockBuilder blockBuilder, IdentifierGenerator idGenerator, NewArrayTree tree) {
     Expression.Variable newArray = variableWithId(idGenerator.newIdFor(tree));
     blockBuilder.newObject(newArray, tree.symbolType().fullyQualifiedName(), location(tree));
     idGenerator.varForExpression(tree, newArray.id());
-
-    for (ExpressionTree initializer : tree.initializers()) {
-      Expression argument = idGenerator.lookupExpressionFor(initializer);
-      blockBuilder.assignTo(variableWithId(idGenerator.newId()), arraySet(newArray, argument), location(tree));
+    List<ExpressionTree> initializers = tree.initializers();
+    if (!initializers.isEmpty() && isObject(initializers.get(0).symbolType())) {
```

Comment:
why checking only the first element?
```java
  void foo(String tainted) {
    Object[] array = {0, tainted};
    String query = String.format("SELECT * FROM %1 LIMIT %0", array);
    // ...
  }
```
For mixed type arrays, you may want to check every initializers